drivers/arm/ccn/ccn.c

```diff
@@ -490,3 +490,79 @@ int ccn_get_part0_id(uintptr_t periphbase)
 	return (int)(mmio_read_64(periphbase
 			+ MN_PERIPH_ID_0_1_OFFSET) & 0xFF);
 }
+
+/*******************************************************************************
+ * This function returns the region id corresponding to a node_id of node_type.
+ ******************************************************************************/
+static unsigned int get_region_id_for_node(node_types_t node_type, unsigned int node_id)
+{
+	unsigned int mn_reg_off, region_id;
+	unsigned long long node_bitmap;
+	unsigned int loc_node_id, node_pos_in_map = 0;
+
+	assert(node_type < NUM_NODE_TYPES);
+	assert(node_id < MAX_RN_NODES);
+
+	/*
+	 * RN-F, RN-I, RN-D , HN-F, HN-I, SN node registers in the MN region occupy
+	 * contiguous 16 byte apart offsets.
+	 */
+	mn_reg_off = MN_RNF_NODEID_OFFSET + (node_type << 4);
+	node_bitmap = ccn_reg_read(ccn_plat_desc->periphbase, MN_REGION_ID, mn_reg_off);
+
+	assert(node_bitmap & (1ULL << (node_id)));
+
+	switch (node_type) {
+	case NODE_TYPE_RNI:
+		region_id = RNI_REGION_ID_START; break;
+	case NODE_TYPE_HNF:
+		region_id = HNF_REGION_ID_START; break;
+	case NODE_TYPE_HNI:
+		region_id = HNI_REGION_ID_START; break;
+	case NODE_TYPE_SN:
+		region_id = SBSX_REGION_ID_START; break;
+	default:
+		ERROR("Invalid Node Type.\n");
+		assert(0);
+		return REGION_ID_LIMIT;
+	}
+
+	FOR_EACH_PRESENT_NODE_ID(loc_node_id, node_bitmap) {
+		INFO("Index = %d with loc_nod=%d and input nod=%d\n", node_pos_in_map, loc_node_id, node_id);
+		if (loc_node_id == node_id)
+			break;
+		node_pos_in_map++;
+	}
+	region_id += node_pos_in_map;
+
+	return region_id;
+}
+
+/*******************************************************************************
+ * This function sets the value 'val' to the register at register_offset from
+ * the base address pointed to by the region_id.
+ * where, region id is mapped to a node_id of node_type.
+ ******************************************************************************/
+void set_node_reg(node_types_t node_type, unsigned int node_id, unsigned int reg_offset, unsigned int val)
+{
+	unsigned int region_id = get_region_id_for_node(node_type, node_id);
+	/* Setting the value of Auxilary Control Register of the Node */
+	ccn_reg_write(ccn_plat_desc->periphbase, region_id, reg_offset, val);
+	VERBOSE("Value written is 0x%x\n", val);
```

Comment:
Do we really need a verbose print? I think you should remove them both here and in ``read_node_reg``. They are quite useless as they are right now anyway, you know the value, it would be more useful if you also printed the address.

Reply:
Accepted. I have updated the VERBOSE.

You are correct, it is useful to print address.